src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items:T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  for (let i = 0; i < items.length; i += 1) {
+    result[String(items[i][key])] = [];
+  }
+
+  for (let i = 0; i < items.length; i += 1) {
+    result[String(items[i][key])].push(items[i]);
+  }
```

Comment:
Let's make our logic simpler, why to iterate array twice if we can do it only once? 🙂